js/oceanex.js

```diff
@@ -804,36 +853,157 @@ module.exports = class oceanex extends Exchange {
+    }
+
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeString (accountsByType, fromAccount, fromAccount);
+        const toId = this.safeString (accountsByType, toAccount, toAccount);
+        let category = 'withdraw';
+        if (fromId === 'exchange' && toId === 'futures') {
+            category = 'deposit';
+        } else if (fromId === 'futures' && toId === 'exchange') {
+            category = 'withdraw';
+        } else {
+            throw new ExchangeError (this.id + ' transfer() allows from spot to future or from future to spot');
+        }
+        const request = {
+            'coinCode': currency['id'],
+            'amount': amount,
+            'category': category,
+        };
+        const response = await this.contractPrivatePostIfcontractFundsTransfer (this.extend (request, params));
+        //
+        //     {
+        //         "code":0,
+        //         "message":"Operation is successful",
+        //         "data": {
+        //             "id": 1,
+        //             "order_num": "contract_a78cce70-4226-44ac-b416-246d506ff167",
+        //             "currency_id": "usd",
+        //             "category": "withdraw",
+        //             "state": "processing",
+        //             "amount": "20.0",
+        //             "created_at": 1578688239,
+        //             "updated_at": 1578688239
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data');
+        return this.parseTransfer (data, currency);
+    }
+
+    parseTransfer (transfer, currency = undefined) {
+        //
+        //     {
+        //         "id": 1,
+        //         "order_num": "contract_a78cce70-4226-44ac-b416-246d506ff167",
+        //         "currency_id": "usd",
+        //         "category": "withdraw",
+        //         "state": "processing",
+        //         "amount": "20.0",
+        //         "created_at": 1578688239,
+        //         "updated_at": 1578688239
+        //     }
+        //
+        const currencyId = this.safeNumber (transfer, 'currency_id');
+        const timestamp = this.safeTimestamp (transfer, 'created_at');
+        const category = this.safeString (transfer, 'category');
+        const status = this.safeString (transfer, 'state');
+        return {
+            'info': transfer,
+            'id': this.safeString (transfer, 'id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'code': this.safeCurrencyCode (currencyId, currency),
+            'amount': this.safeNumber (transfer, 'amount'),
+            'fromAccount': category === 'withdraw' ? 'spot' : 'future',
+            'toAccount': category === 'withdraw' ? 'future' : 'spot',
```

Comment:
We should wrap these ternary conditionals in quotes, otherwise the transpiler fails:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#using-ternary-conditionals

```suggestion
            'fromAccount': (category === 'withdraw') ? 'spot' : 'future',
            'toAccount': (category === 'withdraw') ? 'future' : 'spot',
```